From b0695b29ff142116ad5a14690bbc59d99e4b702e Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Tue, 16 Apr 2013 09:39:26 +0800
Subject: [PATCH 674/678] ENGR00254470-1 mx6q-sd:sata device can't work
 properly on most boards

Revert "ENGR00241595-1 mx6q-sabresd:Enable SATA PHY PDDQ default"

This reverts commit 9cb937432d1dc8d2c31afa5194f1ef0d8bdc0392.

Reasons:
* according to SATA Power mode (in SATA protocol) PHY TX/RX/CLK is
powered down automatically according to SATA controller
power mode SATA port support Disable/Slumber/Partial/Enabled(OOB)

The SATA PHY PDDQ mode is one-shot and recommeded to be used for test,
SATA would not work properly if the PHY PDDQ mode is enabled in U-boot.

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 board/freescale/mx6q_sabresd/mx6q_sabresd.c |   22 ----------------------
 1 files changed, 0 insertions(+), 22 deletions(-)

diff --git a/board/freescale/mx6q_sabresd/mx6q_sabresd.c b/board/freescale/mx6q_sabresd/mx6q_sabresd.c
index c896c93..32f23ac 100644
--- a/board/freescale/mx6q_sabresd/mx6q_sabresd.c
+++ b/board/freescale/mx6q_sabresd/mx6q_sabresd.c
@@ -238,8 +238,6 @@ void board_mmu_init(void)
 #define ANATOP_PLL_PWDN_MASK            0x00001000
 #define ANATOP_PLL_HOLD_RING_OFF_MASK   0x00000800
 #define ANATOP_SATA_CLK_ENABLE_MASK     0x00100000
-#define	PORT_PHY_CTL			        0x178
-#define	PORT_PHY_CTL_PDDQ_LOC		    0x100000
 
 #ifdef CONFIG_DWC_AHSATA
 /* Staggered Spin-up */
@@ -254,11 +252,6 @@ int sata_initialize(void)
 	u32 iterations = 1000000;
 
 	if (sata_curr_device == -1) {
-		/* Make sure that the PDDQ mode is disabled. */
-		reg = readl(SATA_ARB_BASE_ADDR + PORT_PHY_CTL);
-		writel(reg & (~PORT_PHY_CTL_PDDQ_LOC),
-				SATA_ARB_BASE_ADDR + PORT_PHY_CTL);
-
 		/* Reset HBA */
 		writel(HOST_RESET, SATA_ARB_BASE_ADDR + HOST_CTL);
 
@@ -352,17 +345,6 @@ static int setup_sata(void)
 	reg |= 0x59124c6;
 	writel(reg, IOMUXC_BASE_ADDR + 0x34);
 
-    /* FIXME */
-    /*
-     * It needs to wait SATA PHY initialize completed, otherwise write the
-     * PORT_PHY_CTL will fail, then can't enable PDDQ which let PHY entry LPM
-     * Currently set it as 1ms.
-     */
-	__udelay(1000);
-	/* Enable PDDQ mode in default */
-    writel(readl(SATA_ARB_BASE_ADDR + PORT_PHY_CTL) | PORT_PHY_CTL_PDDQ_LOC,
-			SATA_ARB_BASE_ADDR + PORT_PHY_CTL);
-
 	return 0;
 }
 
@@ -1982,10 +1964,6 @@ int checkboard(void)
 		break;
 	}
 
-	if (cpu_is_mx6q())
-		printf("SATA PDDQ: %s\n", ((readl(SATA_ARB_BASE_ADDR
-							+ PORT_PHY_CTL)
-			& PORT_PHY_CTL_PDDQ_LOC)>>20) ? "enabled" : "disabled");
 #ifdef CONFIG_SECURE_BOOT
 	if (check_hab_enable() == 1)
 		get_hab_status();
-- 
1.7.7.4

